src/pudl/transform/ferc1.py

```diff
@@ -6509,3 +6540,135 @@ def plants_steam_ferc1(
         transformed_fuel=fuel_ferc1,
     )
     return convert_cols_dtypes(df, data_source="ferc1")
+
+
+def get_other_dimensions() -> list[str]:
+    """Get a list of the other dimension columns across all of the transformers."""
+    # grab all of the dimensions columns that we are currently verifying as a part of
+    # reconcile_table_calculations
+    other_dimensions = []
+    for transformer in FERC1_TFR_CLASSES.values():
+        other_dimensions.append(
+            transformer().params.reconcile_table_calculations.subtotal_column
+        )
+    # remove nulls and dedupe
+    other_dimensions = [sub for sub in other_dimensions if sub]
+    other_dimensions = list(set(other_dimensions))
+    return other_dimensions
+
+
+def table_to_xbrl_factoid_name() -> dict[str, str]:
+    """Build a dictionary of table name (keys) to ``xbrl_factiod`` column name."""
+    return {
+        table_name: transformer().params.xbrl_factoid_name
+        for (table_name, transformer) in FERC1_TFR_CLASSES.items()
+    }
+
+
+@asset(ins={table_name: AssetIn(table_name) for table_name in FERC1_TFR_CLASSES})
+def table_dimensions_ferc1(**kwargs) -> pd.DataFrame:
+    """Build a table of dimensions found in the transformed data tables.
+
+    The dimensions that show up within the transformed tables differ across table and
```

Comment:
I am also not really understanding.

Reply:
yea i had trouble describing this so your not off base for not following. it builds a table with columns of: table_name + xbrl_factoid + all the other dimension columns ( i think its just utility_type, plant_function rn). the data in the table represent all of the other dimensions that are present in the tables.

(as an aside I sometimes want to type hints with column name for dataframes bc i think that would be generally helpful. I think pandera could be used to do this and i kinda wanna figure that out)

Reply:
I'm still not getting your explanation. What does each record in the table being compiled represent?  When you say "The data in the table represent all of the other dimensions that are present in the tables" do you mean there's something other than the dimension columns in the table that's being compiled?

Or rather than talking about something outside of the dimensions, are you looking at the **values** which appear in each dimension, for each table?  Maybe...

"Compile a dataframe indicating what distinct values are observed in the data for each dimension column in association with each unique combination of `table_name` and `xbrl_factoid`. E.g. for all factoids found in the `plant_in_service_ferc1` table, the only value observed for `utility_type` is `electric` and the only value of `plant_status` observed is `in_service`." (Though this might be a bad example because I think both `utility_type` and `plant_status` are implicit in the `plant_in_service_ferc1` table so maybe they don't actually end up in this compilation? Or have they already been added into the data at this point?)

(This does seem to be exactly what Pandera is for and I agree it would be super helpful. Just saying "it's a dataframe" is not a whole lotta info.)